initfiles/bash/etc/systemd/system/hpcc-conf@instance.service.in

```diff
@@ -6,10 +6,12 @@ After=network.target remote-fs.target
 Type=oneshot
 User=@RUNTIME_USER@
 Group=@RUNTIME_GROUP@
+ExecStartPre=/bin/mkdir -p @RUNTIME_PATH@/%i @LOG_PATH@/%i @PID_PATH@ @LOCK_PATH@/%i
+ExecStartPre=/bin/chown -R @RUNTIME_USER@:@RUNTIME_GROUP@ @RUNTIME_PATH@ @LOG_PATH@ @PID_PATH@ @LOCK_PATH@
 ExecStart=@ADMIN_PATH@/check-component-exists.sh %i
-ExecStart=/bin/mkdir -p @RUNTIME_PATH@/%i @LOG_PATH@/%i @PID_PATH@ @LOCK_PATH@/%i
 ExecStart=@ADMIN_PATH@/configgen -env @CONFIG_DIR@/@ENV_XML_FILE@ -od @RUNTIME_PATH@ -id @COMPONENTFILES_PATH@/configxml -c %i
 ExecStart=/bin/bash -c '/bin/echo "PATH=$PATH:@EXEC_PATH@:@ADMIN_PATH@:@RUNTIME_PATH@/%i" > @RUNTIME_PATH@/%i/%i.cfg'
+PermissionsStartOnly=true
```

Comment:
Causes ExecStartPre to run as root, and not as hpcc user/group.

ExecStartPre lays down the paths if not present, and then recursively makes hpcc:hpcc the owner of all the contents. This will insure that on systems with /var/run/ /var/lock as tmpfs, they will be created appropriately after a reboot.